### PR TITLE
Fix openURL freeze application start

### PR DESCRIPTION
**Pull request by clpolet**

### CODE REVIEW

File: MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m [hunk 1]
```diff
@@ -383,7 +383,11 @@ - (void)openURLInEmbeddedSafari:(NSURL *)url fromClass:(Class)clazz {
 
 - (void)openURLInSafariApp:(NSURL *)url {
   MSLogDebug([MSDistribute logTag], @"Using Safari browser to open URL: %@", url);
-  [MSUtil sharedAppOpenUrl:url options:@{} completionHandler:nil];
+  
+  /* Dispatch the open url call to the next loop to avoid freezing the App new instance start up */
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [MSUtil sharedAppOpenUrl:url options:@{} completionHandler:nil];
```
Comment (jaelim-ms) [inline MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m]: Isn't it better to have it in `sharedAppOpenUrl`?
What if it is called from main thread? You don't need to check current thread?

Comment (clpolet) [inline MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m]: Right I can move it to "sharedAppOpenUrl" for future use. 
The only thing we have to make sure is it's not using openURL in the same loop than the init one which is what we are doing here.

Comment (jaelim-ms) [inline MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m]: Then you can add TODO comment there.

**Comment by clpolet** [comment]
Forgot to push the small change yesterday @jaelim-ms :)

**Comment by jaelim-ms** [comment]
Not unit test is failing. Looks the previous implementation is wrong.

**Pull request merged**